homeassistant/components/camera/__init__.py

```diff
@@ -345,10 +402,29 @@ def state(self):
             return STATE_STREAMING
         return STATE_IDLE
 
+    def turn_off(self):
+        """Turn off camera."""
+        raise NotImplementedError()
+
+    @callback
+    def async_turn_off(self):
+        """Turn off camera."""
+        return self.hass.async_add_job(self.turn_off)
+
+    def turn_on(self, option=None):
+        """Turn off camera."""
+        raise NotImplementedError()
+
+    @callback
+    def async_turn_on(self, option=None):
+        """Turn off camera."""
+        return self.hass.async_add_job(self.turn_on, option)
+
     def enable_motion_detection(self):
         """Enable motion detection in the camera."""
         raise NotImplementedError()
 
+    @callback
```

Comment:
This is wrong. This function returns a task that has to be awaited.

Reply:
You mean I should not add callback, but should change the method signature to `async def`?

Reply:
The function used to return a task that needs to be awaited.

By labelling it callback, you tell people not to await it, meaning the function is never executed.

If you change it to async, it's more clear, but then you also need to add an await.